added documentation for the text-decoration-control component (#57184)
WordPress/gutenberg: packages/block-editor/src/components/text-decoration-control/README.md

# TextDecorationControl

> **Notice: This feature is still experimental. “Experimental” means this is an early implementation subject to drastic and breaking changes.**

Reviewer: ```suggestion
<div class="callout callout-alert">
This feature is still experimental. “Experimental” means this is an early implementation subject to drastic and breaking changes.
</div>
<br />
```

I believe these readme files can be pulled into other documentation such as the Gutenberg Handbook. In those cases, using the `div` with callout classes helps highlight this important note.

You can see this already in use for the [Block Variation Picker component](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/components/block-variation-picker/README.md?plain=1#L3C1-L5).

![TextDecorationControl Element in Inspector Control](https://raw.githubusercontent.com/WordPress/gutenberg/HEAD/docs/assets/text-decoration-component.png?raw=true)

## Usage

```jsx

> import { __experimentalTextDecorationControl } from '@wordpress/block-editor';

Reviewer: ```suggestion
import { __experimentalTextDecorationControl as TextDecorationControl } from '@wordpress/block-editor';
```

The experimental component can be renamed to `TextDecorationControl` and updated below as well. This would  help with the readability of the example and minimise what would need to be updated when the component is stabilized.

```

Then, you can use the component in your block editor UI:

```jsx

> <__experimentalTextDecorationControl

Reviewer: ```suggestion
<TextDecorationControl
```
If renaming the experimental component as suggested above, we'll also need to update the example here.

# TextDecorationControl

![TextDecorationControl Element in Inspector Control](https://raw.githubusercontent.com/WordPress/gutenberg/HEAD/docs/assets/text-decoration-component.png?raw=true)

## Table of Contents

1. [Usage](#usage)

> 3. [Properties](#props)

Reviewer: I'm not sure we need the ToC given its a small readme and other readmes don't appear to include it.

> ## Usage

Reviewer: It might be best if we still with a single code block for the usage example to be consistent with other readmes, such as the one you added for `TextTransformControl`.

P.S. That readme also has the same issues with its import statement and it being an experimental component.

```jsx

> import { TextDecorationControl } from '@wordpress/block-editor';

Reviewer: This import statement is incorrect as `TextDecorationControl` is [exported](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/components/index.js?plain=1#L51C1-L52) as an experimental component: `__experimentalTextDecorationControl`.
